dockstore-webservice/src/test/java/io/dockstore/webservice/helpers/DiagnosticsHelperTest.java

```diff
@@ -0,0 +1,165 @@
+                }
+                if (throwable != null) {
+                    throw new RuntimeException("unexpected throwable");
+                }
+                output = messagePattern;
+            }
+            @Override
+            protected String getFullyQualifiedCallerName() {
+                return "";
+            }
+            @Override
+            public boolean isErrorEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isErrorEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isWarnEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isWarnEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isInfoEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isInfoEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isDebugEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isDebugEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isTraceEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isTraceEnabled() {
+                return true;
+            }
+        };
+        helper = new DiagnosticsHelper(logger);
+    }
+
+    @Test
+    void testCensoringWithExamples() {
+        confirmPassed("7236463");  // short decimal string
+        confirmPassed("af081afb609d");  // short hex string
+        confirmPassed("K11/sjsbWU3+");  // short base64-ish string
+        confirmPassed("/This/Is/A/Test/Of/The/Censoring/System");  // path that contains only base64 characters
+        confirmCensored("f893b559e572f323b5369452752d56743970e39132136aff378b76492c7549dd");
+        confirmCensored("039832784664398210947231904282437420197091837413209843");
+        confirmCensored("Lp0+12I8Xlqhi18KDzzdXFUDrSJWV8GxwmivwQf9thRI1/k8Ec3G4t7Hxoz8fEgG");
+        confirmCensored("Qz2OJicD0w__J__deT42-DjtfrTG3ZOD0rP0PchWyPQXVpL96sXJQYm");
+        confirmCensored("KmqLbmoBq8cuQUu8rwIVXCSyafM+oakMEqf3z75Cr9");
+        Assertions.assertTrue(log("a string with QnDn/5VCnYI7hbeM9Xet9zYSFV2GaiTI7TmXUqE/2ljDtvFVbRB7CRWMqQz+ embedded").endsWith(
+            "a string with XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX embedded"));
+        Assertions.assertTrue(log("fYEJEaL4oECNDCwPOwZHqBlC4+3BjF3Xc4XxDhM two! d3MNbsl4ubApX5ZMoWMm4XsH7No7Q/qoBoQym8M").endsWith(
+            "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX two! XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"));
+    }
+
+    @Test
+    void testCensoringWithRandomBase64Strings() {
+        Random random = new Random(0);
```

Comment:
This test is deterministic, `Random `will generate the same stream of values given a particular seed, and the argument to the constructor (`0`) is the seed.

Reply:
Neat. 
TIL `The legacy classes java.util.Random and java.util.SecureRandom were backfitted to implement this interface, so if you switch the declaration to the new interface, the legacy implementations become interchangeable with the new generators. The legacy classes’ algorithms were not improved, however, to avoid breaking Java’s ever-important backwards compatibility. In fact, given the same seed, the java.util.Random class in both Java 11 and Java 17 will output exactly the same sequence.`
https://blogs.oracle.com/javamagazine/post/java-pseudo-random-number-generator-enhancements